scripts/mutect2_wdl/mutect2.wdl

```diff
@@ -22,32 +22,57 @@ task M2 {
   File tumor_bam
   File tumor_bam_index
   String tumor_sample_name
-  File normal_bam
-  File normal_bam_index
-  String normal_sample_name
+  File? normal_bam
+  File? normal_bam_index
+  String? normal_sample_name
   File? pon
   File? pon_index
   File? dbsnp
   File? dbsnp_index
   File? cosmic
   File? cosmic_index
+  String output_vcf_name
 
   command {
+  if [[ "$normal_bam" == *.bam ]]; then
+      normal_command_line="-I ${normal_bam} -normal ${normal_sample_name}"
+  fi
+
     java -Xmx4g -jar ${gatk4_jar} Mutect2 \
     -R ${ref_fasta} \
     -I ${tumor_bam} \
-    -I ${normal_bam} \
     -tumor ${tumor_sample_name} \
-    -normal ${normal_sample_name} \
+    $normal_command_line \
     ${"--dbsnp " + dbsnp} \
     ${"--cosmic " + cosmic} \
     ${"--normal_panel " + pon} \
     -L ${intervals} \
-    -O ${tumor_sample_name}-vs-${normal_sample_name}.vcf
+    -O "${output_vcf_name}.vcf"
+  }
+
+  output {
+    File output_vcf = "${output_vcf_name}.vcf"
+  }
+}
+
+# HACK: cromwell can't handle the optional normal sample name in output or input --
+# string interpolation of optionals only works inside a command block
+# thus we use this hack
+task MakeOutputVcfName {
```

Comment:
Please make the ``normal_bam`` parameter a String ... Otherwise, you will localize a bam file on the cloud.

Alternatively, since normal bam and normal sample name are usually both blank or both populated, can't you just ditch the normal_bam parameter entirely?  Then just use normal_sample_name if that is populated?